pagopa/io-app
# feat: [IOCOM-834] Auto refresh on new messages home

⚠️ This PR depends on https://github.com/pagopa/io-app/pull/5940 ⚠️

## Short description
This PR adds the automatic retrieval of new messages in the new messages home.

[<video src="" />](https://github.com/pagopa/io-app/assets/5150343/d0a295b6-0297-4968-a566-c379955edddd)

## List of changes proposed in this pull request
- `allPaginated` reducer adds the `lastRequestTime: Date` to both `archive` and `inbox`, initially set to `new Date(0)`
- actions of type `reloadAllMessages.success` and `loadNextMessagePage.success` update such property with current receiving date
- `getLoadPreviousPageMessagesActionIfAllowed` in `homeUtils` checks, for the shown category (inbox or archive) if the related `lastRequestTime` plus an interval (60 seconds) has expired and, if a `previous` index has been set and the system is not performing any message home asynchronous operation, return the `loadNextMessagePage.request` action
- `PagerViewContainer` calls the method above and dispatches the action, if defined. It is called upon `useFocusEffect` (that maps a navigation event) and when the shown page changes (inbox or archive)

- There is also a new action, `requestAutomaticMessagesRefresh`, that allows any point in the application to schedule for a new message check (upon re-entering the new message home)

## How to test
Using the io-dev-api-server, generate some messages and make sure to enable both DS and new Home. The refresh timeout is 60 seconds so lower it to something like 10 to ease the testing. After the inbox has been loaded, you can use the io-dev-api-server main page to generate a new message (in the home category). At this point:
- switch to the Archive tab, wait for the timeout and switch back to the inbox. The new message should appear
- navigate to any message details, wait for the timeout and go back to the message home. The new message should appear
